### PR TITLE
Add option to set TLS options in-memory for API client

### PR DESCRIPTION
**Pull request by michelvocks**
This PR adds the option to set in-memory certificates to the API client instead of requiring the certificate to be stored on disk in a file.

This allows us to define API client TLS options per Consul secret backend in Vault.
Related issue https://github.com/hashicorp/vault/issues/4800

**Note:** My `go mod vendor` produced a few more deletes than expected. Happy to revoke these if necessary.

### CODE REVIEW

**Comment by i0rek** [comment]
Consul has a thing to update the vendor folder: `make update-vendor`. If you are using this, CI will test your PR.

File: api/api.go [hunk 1]
```diff
@@ -464,12 +476,25 @@ func SetupTLSConfig(tlsConfig *TLSConfig) (*tls.Config, error) {
 			return nil, err
 		}
 		tlsClientConfig.Certificates = []tls.Certificate{tlsCert}
+	} else if tlsConfig.CertFile != "" || tlsConfig.KeyFile != "" {
+		return nil, fmt.Errorf("both client cert and client key must be provided")
+	}
+
+	if len(tlsConfig.CertPEM) != 0 && len(tlsConfig.KeyPEM) != 0 {
```
Comment (i0rek) [inline api/api.go]: Right now `CertPEM` takes precedence over `CertFile`. I don't have a strong opinion, but I think it would better the other way around.

Comment (michelvocks) [reply api/api.go]: Thanks! Fixed.

**Comment by michelvocks** [comment]
@i0rek The failing tests seem to be unrelated to the PR changes. Let me know if you need anything else.

**Pull request merged**